lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1191,31 +1195,22 @@ def _static_scrape_config(self, relation) -> list:
         if not scrape_metadata:
             return scrape_jobs
 
-        job_name_prefix = "juju_{}_prometheus_scrape".format(
-            JujuTopology.from_dict(scrape_metadata).identifier
-        )
+        topology = JujuTopology.from_dict(scrape_metadata)
+
+        job_name_prefix = "juju_{}_prometheus_scrape".format(topology.identifier)
         scrape_jobs = PrometheusConfig.prefix_job_names(scrape_jobs, job_name_prefix)
         scrape_jobs = PrometheusConfig.sanitize_scrape_configs(scrape_jobs)
 
         hosts = self._relation_hosts(relation)
 
         scrape_jobs = PrometheusConfig.expand_wildcard_targets_into_individual_jobs(
-            scrape_jobs, hosts, JujuTopology.from_dict(scrape_metadata)
+            scrape_jobs, hosts, topology
         )
 
         return scrape_jobs
 
-    def _relation_hosts(self, relation) -> dict:
-        """Fetch unit names and address of all metrics provider units for a single relation.
-
-        Args:
-            relation: An `ops.model.Relation` object for which the unit name to
-                address mapping is required.
-
-        Returns:
-            A dictionary that maps unit names to unit addresses for
-            the specified relation.
-        """
+    def _relation_hosts(self, relation: Relation) -> Dict[str, Tuple[str, str]]:
+        """Returns a mapping from unit names to (address, path) tuples, for the given relation."""
```

Comment:
I know our linting is no checking this, but just in case some day we decide to check it, we may get:

`D401: First line should be in imperative mood (perhaps 'Return', not 'Returns') [D401]`

Reply:
That sounds like a bug in the linter, or it's interpreting a single line as a Google-format docstring for some reason. Maybe you should open a bug on the linter that reported it, because "Returns" is the correct English here. I doubt if they're going to pull in NLTK to try to get English right, though ;)

Reply:
https://github.com/canonical/observability-libs/issues/27